sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -152,10 +151,11 @@ case class CreateViewCommand(
             sparkSession,
             analyzedPlan,
             aliasedPlan.schema,
-            originalText,
-            child))
+            originalText))
       } else {
-        aliasedPlan
+        TemporaryViewRelation(
+          prepareTemporaryViewFromDataFrame(name, aliasedPlan),
```

Comment:
Here is a mistake: in the `if` branch, we call `prepareTemporaryView(viewIdent, ...`, and here we should also use `viewIdent` instead of `name`. Global temp view has 2-part name and the `name` may not be qualified (e.g. `df.createGlobalTempView`).

@imback82 can you help to fix it with a test?

Reply:
good catch. will fix.

Reply:
Created https://github.com/apache/spark/pull/31783. Thanks!